pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -163,3 +164,75 @@ def brick_sort(array, **kwargs):
 
     if _check_type(array, DynamicArray):
         array._modify(force=True)
+
+def _brick_sort_comp(array, i, j, comp):
+    if(_comp(array[j], array[i], comp)):
+        array[i], array[j] = array[j], array[i]
+        return False
+    return True
```

Comment:
```suggestion
def _brick_sort_comp(array, i, j, comp, is_sorted):
    is_sorted = _comp(array[i], array[j], comp)
    if not is_sorted:
        array[i], array[j] = array[j], array[i]
```

Reply:
wouldnt this cause race condition since all the threads will try to access is_sorted

Reply:
Well, we want to run the next pass only when at least a single swap occurs in the current pass. This is what it represents.

Reply:
got it. I thought that `result()` works like `join()` and will wait till the function returns a result.